### ISSUE TITLE
flameshot aborts screenshots when switching tools from text

### ISSUE THREAD

**Issue by karlyeurl** [flameshot-org/flameshot#2429]
### Flameshot Version

```
$ flameshot --version
Flameshot v11.0.0 (-)
Compiled with Qt 5.15.2

```

### Installation Type

Linux, MacOS, or Windows Package manager (apt, pacman, eopkg, choco, brew, ...)

### Operating System type and version

Linux

### Description

When running `flameshot gui` and selecting an area, if we are writing text with the text tool, switching tools causes flameshot to abort and a notification to pop: "**Flameshot Info** Screenshot Aborted".

This does not happen if we deselect the text tool first.

I believe this is a bug, as it was not happening in the prior version of flameshot.

### Steps to reproduce

1. run `flameshot gui`
2. select an area
3. select the text tool
4. click in the area and start writing text
5. without unselecting the text tool, pick another tool (I used the "circle" paint tool, but most other tools trigger this too)
6. `flameshot gui` exits with return code `0`, a system notification is sent with "**Flameshot Info** Screenshot Aborted".

### System Information

1. linux 5.16.9-arch1-1
2. two screens
3. awesome-wm with xorg

**Comment by mmahmoudian** [comment]
Isn't this duplicate of #2283, #2353, #2402, and #2407?

**Comment by karlyeurl** [comment]
Oh for the love of God, I'm an idiot. I searched in _open_ issues.

Sorry! Closing.

**Comment by dogunbound** [comment]
Has this bug been fixed? I'm still having it.

**Comment by mmahmoudian** [comment]
@dogunbound read my last comment (the one before this one)

**Comment by dogunbound** [comment]
@mmahmoudian  oh lol. Thanks.

**Comment by mo** [comment]
Yes I saw it. I think it makes sense to fix it in Ubuntu 22.04 as well, many people are upgrading to it right now and on aug 4th they will turn on automatic upgrade of all LTS users to this new Ubuntu version. I installed the latest dev version now, so it works well for me.
